optional-container-engine/config.py

```diff
@@ -0,0 +1,80 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This file contains all of the configuration values for the application.
+Update this file with the values for your specific Google Cloud project.
+You can create and manage projects at https://console.developers.google.com
+"""
+
+# The secret key is used by Flask to encrypt session cookies.
+SECRET_KEY = 'secret'
+
+# There are three different ways to store the data in the application.
+# You can choose 'datastore', 'cloudsql', or 'mongodb'. Be sure to
+# configure the respective settings for the one you choose below.
+# You do not have to configure the other data backends. If unsure, choose
+# 'datastore' as it does not require any additional configuration.
+DATA_BACKEND = 'datastore'
+
+# Google Cloud Project ID. This can be found on the 'Overview' page at
+# https://console.developers.google.com
+PROJECT_ID = 'your-project-id'
+
+# Cloud Datastore dataset id, this is the same as your project id.
+DATASTORE_DATASET_ID = PROJECT_ID
+
+# SQLAlchemy configuration
+# Replace user, pass, host, and database with the respective values of your
+# Cloud SQL instance.
+SQLALCHEMY_DATABASE_URI = \
+    'mysql+pymysql://user:password@host/database'
+
+# Mongo configuration
+# If using mongolab, the connection URI is available from the mongolab control
+# panel. If self-hosting on compute engine, replace the values below.
+MONGO_URI = \
+    'mongodb://user:password@host:27017/database'
+
+# Google Cloud Storage and upload settings.
+# Typically, you'll name your bucket the same as your project. To create a
+# bucket:
+#
+#   $ gsutil mb gs://<your-bucket-name>
+#
+# You also need to make sure that the default ACL is set to public-read,
+# otherwise users will not be able to see their upload images:
+#
+#   $ gsutil defacl set public-read gs://<your-bucket-name>
+#
+# You can adjust the max content length and allow extensions settings to allow
+# larger or more varied file types if desired.
+CLOUD_STORAGE_BUCKET = 'your-project-id'
+MAX_CONTENT_LENGTH = 8 * 1024 * 1024
+ALLOWED_EXTENSIONS = set(['png', 'jpg', 'jpeg', 'gif'])
+
+# OAuth2 configuration.
+# This can be generated from the Google Developers Console at
+# https://console.developers.google.com/project/_/apiui/credential.
+# Note that you will need to add all URLs that your application uses as
+# authorized redirect URIs. For example, typically you would add the following:
+#
+#  * http://localhost:8080/oauth2callback
+#  * https://<your-app-id>.appspot.com/oauth2callback.
+#
+# If you receive a invalid redirect URI error review you settings to ensure
+# that the current URI is allowed.
+GOOGLE_OAUTH2_CLIENT_ID = \
```

Comment:
This actually isn't used in this tutorial (or the GCE one) as OAuth2 requires a domain name which is a significant amount of setup.